src/js/cookie-banner.js

```diff
@@ -1,27 +1,34 @@
 (function() {
   const cookieBanner = document.getElementById('cookiebanner');
   const cookieBannerWrapper = document.getElementById('nhsuk-cookie-banner');
-  const cookieAcceptButton = document.getElementById('nhsuk-cookie-banner__link_accept_analytics');
+  const cookieAcceptButton = document.getElementById('nhsuk-cookie-banner__link_accept');
+  const cookieChooseButton = document.getElementById('nhsuk-cookie-banner__link');
   const cookieConfirmationBanner = document.getElementById('nhsuk-cookie-confirmation-banner');
-  const cookieRejectButton = document.getElementById('nhsuk-cookie-banner__link_accept');
 
-  function handleLinkClick(consent) {
+  function handleAcceptButtonClick() {
     cookieBanner.style.display = 'none';
     cookieConfirmationBanner.style.display = 'block';
-    setConsentCookie({
-      necessary: true,
-      preferences: false,
-      statistics: consent,
-      marketing: false,
-      consented: true,
-    });
+    setCookiePreferences((hasSeenBanner = true), (hasAcceptedOptionalCookies = true));
     startAnalytics();
   }
 
-  if (!getConsentCookie()) {
+  function handleChooseButtonClick() {
+    cookieBanner.style.display = 'none';
+    setCookiePreferences((hasSeenBanner = true));
+    window.location.assign('https://access.login.nhs.uk/cookies/cookie-settings');
+  }
+
+  const cookiePreferences = getCookiePreferences();
+  setCookiePreferences();
+
+  if (!cookiePreferences.hasSeenBanner) {
     cookieBannerWrapper.style.display = 'block';
-    cookieAcceptButton.addEventListener('click', () => handleLinkClick(true));
-    cookieRejectButton.addEventListener('click', () => handleLinkClick(false));
+    cookieAcceptButton.addEventListener('click', () => handleAcceptButtonClick());
+    cookieChooseButton.addEventListener('click', () => handleChooseButtonClick());
+
+    setTimeout(function() {
+      cookieBannerWrapper.focus();
+    }, 500);
```

Comment:
any reason we need to delay for 500ms?

Reply:
Yea, it's not ideal honestly. It's there to scroll the page back up to the cookie banner but if the timeout was shorter it didn't work. Might be better to watch the DOM for the banner becoming visible and react to that?

Reply:
Had a quick look at the code for that and it seems a bit overkill! This is fine, just maybe add a comment?

Reply:
Cool. Will add the comment.